tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/EventBridge/ContextPropagation.cs

```diff
@@ -0,0 +1,107 @@
+            where TPutEventsRequest : IPutEventsRequest, IDuckType
+        {
+            var entries = request.Entries.Value;
+            if (entries == null)
+            {
+                return;
+            }
+
+            foreach (var entry in entries)
+            {
+                var duckEntry = entry.DuckCast<IPutEventsRequestEntry>();
+                if (duckEntry != null)
+                {
+                    InjectHeadersIntoDetail(duckEntry, context);
+                }
+            }
+        }
+
+        // Tries to add Datadog trace context under the `_datadog` key at the top level of the `detail` field.
+        // `detail` is a string, so we have to manually modify it using a StringBuilder.
+        private static void InjectHeadersIntoDetail(IPutEventsRequestEntry entry, SpanContext context)
+        {
+            var detail = entry.Detail?.Trim() ?? "{}";
+            if (!detail.EndsWith("}"))
+            {
+                // Unable to parse detail string, so just leave it unmodified. Don't inject trace context.
+                Log.Information("Unable to parse detail string. Not injecting trace context.");
+                return;
+            }
+
+            var detailBuilder = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize).Append(detail);
+            detailBuilder.Remove(detail.Length - 1, 1); // Remove last bracket
+            if (detail.Length > 2)
+            {
+                detailBuilder.Append(','); // Add comma if the original detail is not empty
+            }
+
+            var traceContext = BuildTraceContextJson(context, entry.EventBusName);
+            detailBuilder.Append($"\"{DatadogKey}\":{traceContext}").Append('}');
+
+            // Check new detail size
+            var updatedDetail = Util.StringBuilderCache.GetStringAndRelease(detailBuilder);
+            var byteSize = Encoding.UTF8.GetByteCount(updatedDetail);
+            if (byteSize >= MaxSizeBytes)
+            {
+                Log.Information("Payload size too large to pass context");
+                return;
+            }
+
+            entry.Detail = updatedDetail;
+        }
+
+        // Builds a JSON string containing Datadog trace context
+        private static string BuildTraceContextJson(SpanContext context, string? eventBusName)
+        {
+            // Inject trace context
+            var jsonBuilder = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
+            jsonBuilder.Append('{');
+            SpanContextPropagator.Instance.Inject(context, jsonBuilder, new StringBuilderCarrierSetter());
+
+            // Inject start time and bus name
+            var startTime = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+            jsonBuilder.Append($"\"{StartTimeKey}\":\"{startTime}\"");
+            if (eventBusName != null)
+            {
+                jsonBuilder.Append($",\"{ResourceNameKey}\":\"{eventBusName}\"");
```

Comment:
Should we use a json serializer instead of doing this by hand? IS there the possibility of characters that need to be escaped?

Reply:
Yes we definitely can use a JSON serializer. Is there a way to do it here without extra overhead? I ran into the same issue on `dd-trace-java`, and the team wanted me to do it by hand instead

Reply:
> The current SNS trace injection also does it manually

Yeah, and I have the same concern there 😅 

> I ran into the same issue on `dd-trace-java`, and the team wanted me to do it by hand instead

Performance is important, but there's a balance to be made between performance and correctness. I'd say if you are confident that no characters here need to be encoded, then it's fine to do manually. In JSON double-quotes (`"`) and backblashes (`\`) _must_ be encoded. Can `eventBusName` contain either of these characters? If so, maybe we can encode them manually instead of using a full-fledged serializer.

Reply:
I just wanted to make sure we thought about this and it wasn't an oversight.

Reply:
Doing it manually here should be fine. The eventbus can only contain these characters:
> numbers, lower/upper case letters, .,-,_.